ksql-engine/src/main/java/io/confluent/ksql/services/ServiceContextFactory.java

```diff
@@ -48,19 +48,16 @@ public static ServiceContext create(
       final KafkaClientSupplier kafkaClientSupplier,
       final Supplier<SchemaRegistryClient> srClientFactory,
       final ConnectClient connectClient,
-      final SimpleKsqlClient ksqlClient
+      final Supplier<SimpleKsqlClient> ksqlClientSupplier
   ) {
-    final Admin adminClient = kafkaClientSupplier.getAdmin(
-        ksqlConfig.getKsqlAdminClientConfigProps()
-    );
 
     return new DefaultServiceContext(
         kafkaClientSupplier,
-        adminClient,
-        new KafkaTopicClientImpl(adminClient),
+        () -> kafkaClientSupplier
+            .getAdmin(ksqlConfig.getKsqlAdminClientConfigProps()),
         srClientFactory,
-        connectClient,
-        ksqlClient
+        () -> connectClient,
```

Comment:
We're still creating a connect client per request then?

We should either make this lazy, or not have `DefaultServiceContext` accept a supplier for this.  It's misleading to have one-per-call _and_ the supplier.

It would be trivial to change this method to take a `Supplier<ConnectClient>` and change the two places its called to pass a supplier.

Reply:
Creating a connect client is basically free - I wouldn't worry about optimizing this so I would just pass it in without the supplier. 
```
  public DefaultConnectClient(
      final String connectUri,
      final Optional<String> authHeader
  ) {
    Objects.requireNonNull(connectUri, "connectUri");
    this.authHeader = Objects.requireNonNull(authHeader, "authHeader");

    try {
      this.connectUri = new URI(connectUri);
    } catch (URISyntaxException e) {
      throw new KsqlException(
          "Could not initialize connect client due to invalid URI: " + connectUri, e);
    }
  }
```